server/server.go

```diff
@@ -938,12 +940,9 @@ func getToken(md metadata.MD) string {
 
 	// check the HTTP cookie
 	for _, t := range md["grpcgateway-cookie"] {
-		header := http.Header{}
-		header.Add("Cookie", t)
-		request := http.Request{Header: header}
-		token, err := request.Cookie(common.AuthCookieName)
-		if err == nil {
-			tokens = append(tokens, token.Value)
+		token, err := httputil.JoinCookies(common.AuthCookieName, strings.Split(t, ";"))
```

Comment:
Just discovered that cookie parsing in not just `strings.Split(";")` . The cookie value might has quotes etc. [Stackoverflow](https://stackoverflow.com/questions/28262376/parse-cookie-string-in-golang) suggest this:

```golang
header := http.Header{}
    header.Add("Cookie", rawCookies)
    request := http.Request{Header: header}

    fmt.Println(request.Cookies()) 
```